### PR TITLE
Update training index

### LINKED ISSUE #24593: Update Training page
Remove link to Reactor Mesh training.

Add George Tech training.

### PR DESCRIPTION
**Pull request by milljm**
Add George Tech.
Move Reactor Mesh to historical training section.

Closes #24593

### CODE REVIEW

File: modules/doc/content/training/index.md [hunk 1]
```diff
@@ -1,10 +1,13 @@
 # Training
 
 !style! halign=left
-The following links detail any MOOSE related available training on the horizon.
+Upcoming MOOSE related training:
 !style-end!
 
-
+- George Tech (2023 Sept 12-14)
```
Comment (loganharbour) [inline modules/doc/content/training/index.md:7]: Who dis

Comment (loganharbour) [reply modules/doc/content/training/index.md:7]: @rpodgorney

Comment (GiudGiud) [reply modules/doc/content/training/index.md]: Full name? 
[Georgia Institute of Technology](https://www.gatech.edu/)

File: modules/doc/content/training/index.md [hunk 2]
```diff
@@ -1,10 +1,13 @@
 # Training
 
 !style! halign=left
-The following links detail any MOOSE related available training on the horizon.
+Upcoming MOOSE related training:
 !style-end!
 
-
+- Georgia Tech (2023 Sept 12-14)
```
Comment (GiudGiud) [inline modules/doc/content/training/index.md:7]: is this moose framework?

File: modules/doc/content/training/index.md (hunk 1 above)
Comment (loganharbour) [reply modules/doc/content/training/index.md]: this was a joke

File: modules/doc/content/training/index.md [hunk 3]
```diff
@@ -1,10 +1,17 @@
 # Training
 
 !style! halign=left
-The following links detail any MOOSE related available training on the horizon.
+Upcoming MOOSE related training:
 !style-end!
 
+- [Programme-MOOSE Modeling and Simulation for Advanced Reactors (2023 Oct 23-26)](https://ncrcaims.inl.gov/Identity/Account/TrainingRegistration)
```
Comment (GiudGiud) [inline modules/doc/content/training/index.md:7]: what does Programme-MOOSE mean?
is this a NEAMS meeting? NEAMS-training? online?

File: modules/doc/content/training/index.md (hunk 2 above)
Comment (permcody) [reply modules/doc/content/training/index.md]: Yes

**Review by cticenhour — DISMISSED** [review]

**Review by permcody — APPROVED** [review]

**Pull request merged**